api/reference/custom_actions.adoc

```diff
@@ -0,0 +1,383 @@
+              "button_color": "#b4c10f",
+              "button_type": "default",
+              "display": true,
+              "open_url": false,
+              "display_for": "single",
+              "submit_how": "one"
+            },
+            "userid": "admin",
+            "created_on": "2018-01-09T20:41:40Z",
+            "updated_on": "2018-01-09T20:41:40Z",
+            "name": "Ping",
+            "visibility": {
+              "roles": [
+                "_ALL_"
+              ]
+            },
+            "resource_action": {
+              "id": "22",
+              "resource_id": "2",
+              "resource_type": "CustomButton",
+              "created_at": "2018-01-09T20:41:40Z",
+              "updated_at": "2018-01-09T20:41:40Z",
+              "ae_namespace": "SYSTEM",
+              "ae_class": "PROCESS",
+              "ae_instance": "Automation",
+              "ae_attributes": {
+                "request": "general/virtual_machines/ping"
+              }
+            },
+            "enabled": true,
+            "href": "https://172.16.30.129/api/custom_buttons/2"
+          }
+        ],
+        "href": "https://172.16.30.129/api/custom_button_sets/23"
+      }
+    ]
+  },
+  "custom_action_buttons": [
+    {
+      "href": "https://172.16.30.129/api/custom_buttons/1",
+      "id": "1",
+      "guid": "22b5bf4f-d5d6-43af-bc7b-77460d120065",
+      "description": "Analyze Virtual Machine",
+      "applies_to_class": "Vm",
+      "options": {
+        "button_icon": "ff ff-timeline",
+        "button_type": "default",
+        "display": true,
+        "open_url": false,
+        "display_for": "single",
+        "submit_how": "one"
+      },
+      "userid": "admin",
+      "created_on": "2018-01-09T20:40:11Z",
+      "updated_on": "2018-01-09T20:40:11Z",
+      "name": "Analyze",
+      "visibility": {
+        "roles": [
+          "_ALL_"
+        ]
+      }
+    },
+    {
+      "href": "https://172.16.30.129/api/custom_buttons/2",
+      "id": "2",
+      "guid": "e9e77bb3-f52a-4d08-8d58-13923031c6de",
+      "description": "Ping Virtual Machine",
+      "applies_to_class": "Vm",
+      "options": {
+        "button_icon": "ff ff-scsi",
+        "button_color": "#b4c10f",
+        "button_type": "default",
+        "display": true,
+        "open_url": false,
+        "display_for": "single",
+        "submit_how": "one"
+      },
+      "userid": "admin",
+      "created_on": "2018-01-09T20:41:40Z",
+      "updated_on": "2018-01-09T20:41:40Z",
+      "name": "Ping",
+      "visibility": {
+        "roles": [
+          "_ALL_"
+        ]
+      }
+    }
+  ],
+  "actions": [
+    ...
+  ]
+}
+
+----
+
+When custom buttons are defined on a VM, the returned VM resource is extended with custom actions which
+can be triggered via a POST action.
+
+The action name to trigger the action is the lower-cased name of the custom button.
+
+As an example, here's a VM without custom buttons:
```

Comment:
Do you think it might be a good idea to elaborate a little bit more here? At first glance of this example, some of them look like they *could* be custom actions. Perhaps just mention that only the actions defined in the API are returned?